homeassistant/components/switchbot_cloud/__init__.py

```diff
@@ -0,0 +1,67 @@
+"""The SwitchBot via API integration."""
+from asyncio import gather
+from dataclasses import dataclass
+from logging import getLogger
+
+from switchbot_api import Device, Remote, SwitchBotAPI
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_API_KEY, CONF_API_TOKEN, Platform
+from homeassistant.core import HomeAssistant
+
+from .const import DOMAIN
+from .coordinator import SwitchBotCoordinator
+
+_LOGGER = getLogger(__name__)
+PLATFORMS: list[Platform] = [Platform.SWITCH]
+
+
+@dataclass
+class Data:
+    """Data to use in platforms."""
+
+    api: SwitchBotAPI
+    switches: list[Device | Remote]
+
+
+async def async_setup_entry(hass: HomeAssistant, config: ConfigEntry) -> bool:
+    """Set up SwitchBot via API from a config entry."""
+    token = config.data.get(CONF_API_TOKEN)
+    secret = config.data.get(CONF_API_KEY)
+
+    if not token or not secret:
+        _LOGGER.debug("Missing token or secret key")
+        return False
```

Comment:
Don't create a config entry with missing data

Reply:
Resolved by suggestion in the next comment